src/arch/evm/SpokeUtils.ts

```diff
@@ -0,0 +1,335 @@
+import assert from "assert";
+import { BytesLike, Contract, PopulatedTransaction, providers } from "ethers";
+import { CHAIN_IDs } from "../../constants";
+import { Deposit, FillStatus, FillWithBlock, RelayData } from "../../interfaces";
+import {
+  bnUint32Max,
+  BigNumber,
+  toBN,
+  bnZero,
+  chunk,
+  getMessageHash,
+  getRelayDataHash,
+  isDefined,
+  isUnsafeDepositId,
+  isZeroAddress,
+  getNetworkName,
+  paginatedEventQuery,
+  spreadEventWithBlockNumber,
+  toBytes32,
+} from "../../utils";
+
+type BlockTag = providers.BlockTag;
+
+/**
+ * @param spokePool SpokePool Contract instance.
+ * @param deposit V3Deopsit instance.
+ * @param repaymentChainId Optional repaymentChainId (defaults to destinationChainId).
+ * @returns An Ethers UnsignedTransaction instance.
+ */
+export function populateV3Relay(
+  spokePool: Contract,
+  deposit: Omit<Deposit, "messageHash">,
+  relayer: string,
+  repaymentChainId = deposit.destinationChainId
+): Promise<PopulatedTransaction> {
+  const v3RelayData: RelayData = {
+    depositor: toBytes32(deposit.depositor),
+    recipient: toBytes32(deposit.recipient),
+    exclusiveRelayer: toBytes32(deposit.exclusiveRelayer),
+    inputToken: toBytes32(deposit.inputToken),
+    outputToken: toBytes32(deposit.outputToken),
+    inputAmount: deposit.inputAmount,
+    outputAmount: deposit.outputAmount,
+    originChainId: deposit.originChainId,
+    depositId: deposit.depositId,
+    fillDeadline: deposit.fillDeadline,
+    exclusivityDeadline: deposit.exclusivityDeadline,
+    message: deposit.message,
+  };
```

Comment:
Thoughts on breaking this out? I suspect that this data structure will be shared by the SVM counterpart

Reply:
Good call - added here: https://github.com/across-protocol/sdk/pull/957/commits/9afa6ef60a078d6e7b97d4f6154028c7d5648497